scylla/src/transport/cluster.rs

```diff
@@ -60,7 +61,7 @@ pub struct Datacenter {
 
 #[derive(Clone)]
 pub struct ClusterData {
-    pub(crate) known_peers: HashMap<SocketAddr, Arc<Node>>, // Invariant: nonempty after Cluster::new()
+    pub(crate) known_peers: HashMap<NodeAddr, Arc<Node>>, // Invariant: nonempty after Cluster::new()
```

Comment:
Do you propose releasing the constraint that, in order for a Node to be recycled, IPs need to match? I.e. now only Host ID, DC and rack must be the same to recycle a Node?

Reply:
> Do you propose releasing the constraint that, in order for a Node to be recycled, IPs need to match? I.e. now only Host ID, DC and rack must be the same to recycle a Node?

Hm, right, it would indeed be nice to update the IP field if it changes, so I think we should indeed create a new Node instance.

This leaves us with the problem of NodeConnectionPool being recreated. However, I realize now that this problem can be avoided:

1. Add `#[derive(Clone)]` for NodeConnectionPool. Most (if not all) fields are already cloneable, and if not we could put `refiller_handle` and `keepaliver_handle` into an Arc and it should work.
2. Make it possible to update the pool refiller with the new details about the node - in this case we only need to tell it about the new IP. We could do it with an ArcSwap or a tokio broadcast channel.

Reply:
So we drop the important invariant of NodeConnectionPool and PoolRefiller, which is constant IP. In fact we have already decided to do so because of serverless proxy hostname resolution.